pylint/reporters/text.py

```diff
@@ -189,7 +194,13 @@ def on_set_current_module(self, module: str, filepath: Optional[str]) -> None:
 
     def write_message(self, msg: Message) -> None:
         """Convenience method to write a formatted message with class default template"""
-        self.writeln(msg.format(self._template))
+        self_dict = msg._asdict()
+        for key, value in self_dict.items():
+            # pylint: disable=fixme
+            # TODO: Add column to list of attributes to be printed as an empty string
+            if value is None and key in PRINT_AS_EMPTY_STRING:
+                self_dict[key] = ""
+        self.writeln(self._template.format(**self_dict))
```

Comment:
`self.writeln(msg.format(self._template))` has been quite elegant, but there is one drawback. Calling it with an unknown key raises a `KeyError`. Ideally we could check a template and replace invalid values with `""` beforehand. I'm currently having the issue to implement some kind of version check in `vscode-python` as calling `pylint` with the new `msg-template` would raise this error.

```shell
Traceback (most recent call last):
  File "/.../venv-310/bin/pylint", line 33, in <module>
    sys.exit(load_entry_point('pylint', 'console_scripts', 'pylint')())
  File "/.../pylint/__init__.py", line 24, in run_pylint
    PylintRun(sys.argv[1:])
  File "/.../pylint/lint/run.py", line 398, in __init__
    linter.check(args)
  File "/.../pylint/lint/pylinter.py", line 992, in check
    self._check_files(
  File "/.../pylint/lint/pylinter.py", line 1041, in _check_files
    self.add_message(symbol, args=msg)
  File "/.../pylint/lint/pylinter.py", line 1513, in add_message
    self._add_one_message(
  File "/.../pylint/lint/pylinter.py", line 1471, in _add_one_message
    self.reporter.handle_message(
  File "/.../pylint/reporters/text.py", line 202, in handle_message
    self.write_message(msg)
  File "/.../pylint/reporters/text.py", line 192, in write_message
    self.writeln(msg.format(self._template))
  File "/.../pylint/message/message.py", line 94, in format
    return template.format(**self._asdict())
KeyError: 'msg1'
```

Reply:
Shouldn't that just raise a new warning? Something like `config-parse-error` as introduced in #5365. We could emit it on an except for the `KeyError`.

Reply:
That doesn't really solve the issue. At the core of it its backwards incompatibility.

We should be able to use a new template, i.e. with `end_lineno`, in old `pylint` versions and still get a usable result. The way it's now, I need to make sure only to add `end_lineno` if we are using the new pylint version.

If at some point we add another key, this will only repeat. Every time incompatible with old versions.

Reply:
Came up with something. A little hacky, but let me know what you think. I added `if not self._checked_template:` because I thought the additional `if` statement would be better than doing the regex statement each line.